processing/src/main/java/io/druid/granularity/QueryGranularity.java

```diff
@@ -56,6 +72,10 @@ else if(name.equals("NONE"))
     {
       return QueryGranularity.NONE;
     }
+    else if(CALENDRIC_GRANULARITIES.containsKey(name))
+    {
+      return new PeriodGranularity(CALENDRIC_GRANULARITIES.get(name), null, null);
```

Comment:
instead of creating a new object every time, it might make more sense to store constants for each predefined granularity string